src/Platform/TargetPhp/PhpBinaryPath.php

```diff
@@ -31,7 +33,23 @@ private function __construct(
         public readonly string $phpBinaryPath,
         private readonly string|null $phpConfigPath,
     ) {
-        // @todo https://github.com/php/pie/issues/12 - we could verify that the given $phpBinaryPath really is a PHP install
+        if (! file_exists($this->phpBinaryPath)) {
```

Comment:
maybe for testing purposes you may want to be able to create an instance without caring about whether it is valid or not, or for performance reasons in a context where you're sure it is valid. Might be worth having this logic in a static factory? (Could be just `PhpBinaryPath::create()`).

Reply:
Yep that seems reasonable, I'll adjust, thank you